docs/metrics_provider.md

```diff
@@ -1,40 +1,51 @@
 # Metrics provider
 
-The idea about metrics provider is to have a metrics driven approach of
-verifying the operators functionality. An operator reconciles a system where
-several resources exist. As a means of measurement and as a safety net, it is a good
-practise to implement metrics providers that emit metrics about the managed
-system and its resources.
+The idea about metrics providers is to have a metrics driven approach of
+verifying the operators functionality. An operator constantly reconciles the
+current state of the system towards a more desired state eventually. As a means
+of measurement and as a safety net, it is a good practise to implement metrics
+providers that emit metrics about the managed system and its resources.
 
-## Examples
 
-In [`aws-operator` we implement a Prometheus collector](https://github.com/giantswarm/aws-operator/tree/master/service/collector)
-to emit metrics about e.g. VPCs. These metrics are used for verifying each
-tenant cluster has a VPC assigned and no VPC is orphaned without having any
-tenant cluster assigned.
 
-In [`azure-operator` we implement a Prometheus collector](https://github.com/giantswarm/azure-operator/tree/master/service/collector)
-to emit metrics about e.g. ARM Deployments. These metrics are used for alerting
-on deployments being in failed state or stuck in upgrading.
+## Separation
+
+The collector implementation should be separate from the operator project
+itself. As we learned over time, combining operators and collectors may cause
+more problems than it is supposed to solve. See the examples below.
+
 
-In [`operatorkit` we implement a Prometheus collector](https://github.com/giantswarm/operatorkit/tree/master/informer/collector)
+
+## Examples
+
+In Azure we have the [`azure-operator`
+project](https://github.com/giantswarm/azure-operator) and its [separate
+associated collector](https://github.com/giantswarm/azure-collector) to emit
+metrics about e.g. ARM Deployments. These metrics are used for alerting on on
+deployments being in failed state or stuck in upgrading.
+
+In [`operatorkit` we implement a Prometheus
+collector](https://github.com/giantswarm/operatorkit/tree/master/informer/collector)
```

Comment:
AFAICT we don't have informer package anymore. Also AIUI even if we had this package this whole section says it should not be used. I got really confused here.

Reply:
Good point that the informer is gone meanwhile. Though we have the controller collector still which makes sense to have as is. What should we improve on the docs further? What are you particularly confused about?

Reply:
Separation section in this PR says:

> The collector implementation should be separate from the operator project
> itself. 

How then collection packages of operator should be used? In the operator itself or the separate collector project? That's what's puzzles me.

Reply:
The new best practice is how we do it in Azure. Separate projects. When this is still unclear, the docs are useless and we should write them anew.